app/models/user.rb

```diff
@@ -4,21 +4,24 @@ class User < ActiveRecord::Base
   devise :database_authenticatable, :registerable, :confirmable,
          :recoverable, :rememberable, :trackable, :validatable
 
-  has_many :pending_purchases
+  has_many :line_items
+  has_one :cart
+
+  after_save :create_cart
+
+  def create_cart
+    Cart.create!(user: self)
+  end
 
   def add_to_cart(book)
```

Comment:
Okay, last comment, but since I don't see a user_id column on the line_items table I'm guessing that association doesn't actually work anymore unless you go through the cart.